### PR TITLE
[C-2508] Update playback positions to be tracked per user id

### PR DESCRIPTION
**Pull request by Kyle-Shanks**
### Description
Update podcast playback positions to be tracked per user id so that users can log out and log into different accounts with predictable behaviour

### Dragons

N/A

### How Has This Been Tested?

Manually tested on web and mobile

### How will this change be monitored?

N/A

### Feature Flags ###

N/A

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/kj-Update-playback-positions-to-track-user-ids

File: packages/common/src/store/playback-position/sagas.ts [hunk 1]
```diff
@@ -36,11 +42,35 @@ function* setInitialPlaybackPositionState() {
...
+    }
+    yield* put(
+      initializePlaybackPositionState({
+        playbackPositionState: convertedLegacyState
+      })
+    )
+    setLocalStorageItem(
+      PLAYBACK_POSITION_LS_KEY,
+      JSON.stringify(convertedLegacyState)
+    )
```
Comment (amendelsohn) [inline packages/common/src/store/playback-position/sagas.ts:72]: Do we also want to clear out the old key?

File: packages/common/src/store/playback-position/slice.ts [hunk 2]
```diff
@@ -10,16 +10,16 @@ type InitializePlaybackPositionStatePayload = {
 
 type SetTrackPositionPayload = {
   trackId: ID
+  userId?: ID | null
```
Comment (amendelsohn) [inline packages/common/src/store/playback-position/slice.ts:13]: should these accept empty? seems like it's a no-op if we don't have user id, so maybe just don't fire the action in the first place? idk which is better

File: packages/common/src/store/playback-position/slice.ts [hunk 3]
```diff
@@ -31,22 +31,29 @@ const slice = createSlice({
       action: PayloadAction<InitializePlaybackPositionStatePayload>
     ) => {
       const { playbackPositionState } = action.payload
-      state.trackPositions =
-        playbackPositionState.trackPositions ?? state.trackPositions
+      const userIds = Object.keys(playbackPositionState)
+      userIds.forEach((userId) => {
+        state[userId] = playbackPositionState[userId]
+      })
```
Comment (amendelsohn) [inline packages/common/src/store/playback-position/slice.ts:37]: can we do `state = { ...playbackPositionState }` or does the proxy not like that?

File: packages/common/src/store/playback-position/types.ts [hunk 4]
```diff
@@ -10,5 +11,7 @@ export type PlaybackPositionInfo = {
 }
 
 export type PlaybackPositionState = {
-  trackPositions: { [Key in ID]?: PlaybackPositionInfo }
+  [Key in ID]?: {
+    trackPositions: { [Key in ID]?: PlaybackPositionInfo }
+  }
```
Comment (amendelsohn) [inline packages/common/src/store/playback-position/types.ts:16]: Don't need to fix here, but wondering if we should start naming these better to make the types more readable:

```
export type PlaybackPositionState = {
  [UserIdKey in ID]?: {
    trackPositions: { [TrackIdKey in ID]?: PlaybackPositionInfo }
  }
```

**Review by amendelsohn — APPROVED** [review]
Looks great! Nice job :)

File: packages/common/src/store/playback-position/slice.ts (hunk 3 above)
Comment (Kyle-Shanks) [reply packages/common/src/store/playback-position/slice.ts:37]: yea i tried that first, it doesn't like that unfortunately

Comment (amendelsohn) [reply packages/common/src/store/playback-position/slice.ts:37]: dang unlukcy

File: packages/common/src/store/playback-position/slice.ts (hunk 2 above)
Comment (Kyle-Shanks) [reply packages/common/src/store/playback-position/slice.ts:13]: This made it easier for type and not having to check it user existed everywhere this was called if i remember correctly, but I agree it's a bit weird

**Pull request merged**